### PR TITLE
Add MemoryDistributedCacheOptions and use it to configure MemoryCache used by MemoryDistributedCache

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Needs rebase once #332 goes in.

Addresse #329

### CODE REVIEW

File: src/Microsoft.Extensions.Caching.Memory/MemoryCacheServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -80,7 +80,44 @@ public static IServiceCollection AddDistributedMemoryCache(this IServiceCollecti
...
+        /// server scenarios as this cache stores items in memory and doesn't expand across multiple machines.
+        /// For those scenarios it is recommended to use a proper distributed cache that can expand across
+        /// multiple machines.
+        /// </remarks>
+        /// <param name="services">The <see cref="IServiceCollection" /> to add services to.</param>
+        /// <param name="setupAction">
+        /// The <see cref="Action{MemoryDistributedCacheOptions}"/> to configure the provided <see cref="MemoryDistributedCacheOptions"/>.
+        /// </param>
+        /// <returns>The <see cref="IServiceCollection"/> so that additional calls can be chained.</returns>
+        public static IServiceCollection AddDistributedMemoryCache(this IServiceCollection services, Action<MemoryDistributedCacheOptions> setupAction)
```
Comment (JunTaoLuo) [inline src/Microsoft.Extensions.Caching.Memory/MemoryCacheServiceCollectionExtensions.cs]: `AddDistributedMemoryCache` vs `MemoryDistributedCache` we should normalize the naming here. Pick one and go with it.

Comment (Tratcher) [inline src/Microsoft.Extensions.Caching.Memory/MemoryCacheServiceCollectionExtensions.cs]: This is ok. Both names are already used publicly, you can't change it without breaking APIs.

Comment (JunTaoLuo) [inline src/Microsoft.Extensions.Caching.Memory/MemoryCacheServiceCollectionExtensions.cs]: I think it's good to keep things consistent and if we are going to break things, we should break them in 2.0

Comment (Tratcher) [inline src/Microsoft.Extensions.Caching.Memory/MemoryCacheServiceCollectionExtensions.cs]: *shrug* yesterday maybe.

Comment (JunTaoLuo) [inline src/Microsoft.Extensions.Caching.Memory/MemoryCacheServiceCollectionExtensions.cs]: Well we are branching on Monday now so... it's functionally yesterday.

**Comment by JunTaoLuo** [comment]
I'm going to merge as is for now, if we want to change the naming, we can do it afterwards.

**Comment by JunTaoLuo** [comment]
merged in https://github.com/aspnet/Caching/commit/5e73a85ef5e6f1bf61f619c1c4c50437e898e616

**Pull request closed without merging**